x-pack/plugins/observability_solution/logs_data_access/server/services/get_logs_rate_timeseries/get_logs_rate_timeseries.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import type { AggregationOptionsByType, AggregationResultOf } from '@kbn/es-types';
+import { ElasticsearchClient } from '@kbn/core/server';
+import { existsQuery, kqlQuery } from '@kbn/observability-plugin/server';
+import { estypes } from '@elastic/elasticsearch';
+import { RegisterServicesParams } from '../register_services';
+import { getBucketSizeFromTimeRangeAndBucketCount } from '../../utils';
+
+export interface LogsRateTimeseries {
+  esClient: ElasticsearchClient;
+  serviceEnvironmentQuery?: QueryDslQueryContainer[];
+  serviceNames: string[];
+  identifyingMetadata: string;
+  timeFrom: number;
+  timeTo: number;
+  kuery?: string;
+}
+
+interface LogsRateTimeseriesHistogram {
+  timeseries: AggregationResultOf<
+    {
+      date_histogram: AggregationOptionsByType['date_histogram'];
+    },
+    {}
+  >;
+  doc_count: number;
+  key: string;
+}
+
+interface LogRateQueryAggregation {
+  service: estypes.AggregationsTermsAggregateBase<LogsRateTimeseriesHistogram>;
+}
+export interface LogsRateTimeseriesReturnType {
+  [serviceName: string]: Array<{ x: number; y: number | null }>;
+}
+export function createGetLogsRateTimeseries(params: RegisterServicesParams) {
```

Comment:
is it possible to get some test around this one?

Reply:
same here: I wrote a test for the endpoint